src/Filters/design.jl

```diff
@@ -191,9 +201,9 @@ function Elliptic(T::Type, n::Integer, rp::Real, rs::Real)
     # Eq. (65)
     v0 = -im/n*asne(im/εp, k1)
 
-    z = Vector{Complex{T}}(undef, 2*div(n, 2))
-    p = Vector{Complex{T}}(undef, n)
-    gain = one(T)
+    z = Vector{CT}(undef, 2*div(n, 2))
+    p = Vector{CT}(undef, n)
+    gain::T = one(T) # somehow this changes type ...
```

Comment:
I'd prefer if we could figure out/eliminate where this changes type. My guess is that `ze` and `pole` below are `Complex{Float64}`, no matter what `T` is. Maybe it's sufficient to use `z[2i]`/`p[2i]` (or `p[end]`) below when updating `gain`.

Reply:
Following through the code, `cde`, `sne`, and `landen` all promote up to their inputs, and `v0 = -im/n*asne(im/εp, k1)` is `Float64` cause of the `im/n` bit, so that propagates through to `pole` and 
therefore `gain`. I think `ze::T`.
I think using `pole` and `ze` improves legibility, and having an explicit type declaration doesn't really hurt anything at all. But we could force `v0::T` instead, but `convert(T, im/n)` doesn't seem all that different than `gain::T`

Reply:
It makes a substantial difference in the generated code, which you can see even by looking at `@code_warntype`. Stopping the unwanted type promotion at the source, with `convert(T, im/n)` leads to most intermediate variables being of type `T`, while using type annotation with `gain::T` leads most of the intermediate variable to being `Float64`. I think it's cleaner to only use type annotation where absolutely necessary, so I would also vote for using `convert(T, im/n)` over `gain::T`.

Reply:
I did not know that.
I changed it to `im/convert(T, n)`.